pytorch_lightning/trainer/properties.py

```diff
@@ -118,16 +118,16 @@ def is_global_zero(self) -> bool:
 
     @property
     def slurm_job_id(self) -> Optional[int]:
-        try:
-            job_id = os.environ['SLURM_JOB_ID']
-            job_id = int(job_id)
-
-            # in interactive mode, don't make logs use the same job id
-            in_slurm_interactive_mode = os.environ['SLURM_JOB_NAME'] == 'bash'
-            if in_slurm_interactive_mode:
+        job_id = os.environ.get('SLURM_JOB_ID')
+        if job_id:
+            try:
+                job_id = int(job_id)
+            except ValueError:
                 job_id = None
 
-        except Exception:
+        # in interactive mode, don't make logs use the same job id
+        in_slurm_interactive_mode = os.environ.get('SLURM_JOB_NAME') == 'bash'
+        if in_slurm_interactive_mode:
             job_id = None
         return job_id
```

Comment:
We can simplify by catching `TypeError` to avoid `int(None)` and using early return:

```python
        if os.environ.get('SLURM_JOB_NAME') == 'bash':
            # in interactive mode, don't make logs use the same job id
            return None
        try:
            return int(os.environ.get('SLURM_JOB_ID'))
        except (ValueError, TypeError):
            return None
```

Reply:
yeah, I am not much fun of multiple returns unless there is a true advantage, because then you need to find all exits if you want to debug it with checkpoints...